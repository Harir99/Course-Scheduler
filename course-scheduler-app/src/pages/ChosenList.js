import { React } from 'react'
import { Button, Card, ListGroup } from 'react-bootstrap'
import '../index.css'

function ChosenList (props) {
  // Ensuring no duplicate courses
  // (in case of multiple selection of a single section)
  const chosen = []
  props.chosen.forEach((title) => {
    if (!chosen.includes(title)) {
      chosen.push(title)
    }
  })

  // Creating button
  let title
  if (chosen.length > 0) {
    title = (
            <div>
                <hr /> <h4 className="text-purple"> Selected Courses</h4>
            </div>
    )
  } else {
    title = null
  }

  // Function to check if the course is conflicting and changing background color
  function checkConflict (item) {
    let conFlag = 0
    for (let j = 0; j < props.conflict.length; j++) {
      if (item.includes(props.conflict[j].split(' ')[0])) {
        conFlag = 1
      }
    }

    if (conFlag === 1) {
      return (
                <ListGroup.Item key={item} style={{ backgroundColor: 'red', color: 'white' }}>
                    <Card.Body>
                        <Card.Text>
                            {item}
                        </Card.Text>
                            <Button
                                value={item}
                                onClick={props.deleteCourse}
                                size="sm"
                                style={{
                                  backgroundColor: '#84428C',
                                  border: '#84428C'
                                }}
                            >
                                Delete Course
                            </Button>
                    </Card.Body>
                </ListGroup.Item>
      )
    } else {
      return (
                <ListGroup.Item key={item}>
                    <Card.Body>
                        <Card.Text>
                            {item}
                        </Card.Text>
                        <Button
                            value={item}
                            onClick={props.deleteCourse}
                            size="sm"
                            style={{
                              backgroundColor: '#84428C',
                              border: '#84428C'
                            }}
                        >
                            Delete Course
                        </Button>
                    </Card.Body>
                </ListGroup.Item>
      )
    }
  }

  return (
        <div>
            {title}
            <Card className={`text-center ${(props.chosen.length !== 0) ? 'scrollable' : ''}`}>
                <ListGroup variant="flush">
                    {chosen.map((item) => (
                      checkConflict(item)
                    )
                    )}
                </ListGroup>
            </Card>
        </div>

  )
}

export default ChosenList
