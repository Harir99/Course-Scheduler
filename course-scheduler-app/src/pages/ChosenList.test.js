import ChosenList from './ChosenList'
import renderer from 'react-test-renderer'
import React from 'react'

test('tests that chosen list component can be rendered', () => {
  const tree = renderer.create(<ChosenList chosen={['ACCT*1220*0101 (6573) Intro Financial Accounting']} conflict= {[]} setScheduleData={[]} setConflictData={[]} deleteCourse={[]} Semester={'Fall 2022'} setChosenCourses={[]}/>).toJSON()
  expect(tree).toMatchSnapshot()
})
