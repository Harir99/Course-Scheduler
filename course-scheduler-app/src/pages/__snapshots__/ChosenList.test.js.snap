// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests that chosen list component can be rendered 1`] = `
<div>
  <div>
    <hr />
     
    <h4
      className="text-purple"
    >
       Selected Courses
    </h4>
  </div>
  <div
    className="text-center scrollable card"
  >
    <div
      className="list-group list-group-flush"
      onKeyDown={[Function]}
    >
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        <div
          className="card-body"
        >
          <p
            className="card-text"
          >
            ACCT*1220*0101 (6573) Intro Financial Accounting
          </p>
          <button
            className="btn btn-primary btn-sm"
            disabled={false}
            onClick={Array []}
            style={
              Object {
                "backgroundColor": "#84428C",
                "border": "#84428C",
              }
            }
            type="button"
            value="ACCT*1220*0101 (6573) Intro Financial Accounting"
          >
            Delete Course
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
